From d533d240bb5dee1dba474528696343120b768df9 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Wed, 22 May 2024 18:24:31 +0200
Subject: [PATCH 1135/1141] media: stm32: dcmipp: fill media ctl hw_revision
 field

Add hw_revision field information of the media controller so that
application can distinguish between variants of DCMIPP implementations.

Change-Id: I1f709f616765e39ee910f624b3e77285b5f1bfb2
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/381827
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Hugues FRUCHET <hugues.fruchet@st.com>
Reviewed-by: Stephane DANIEAU <stephane.danieau@st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Fabien DESSENNE <fabien.dessenne@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c
index 4faffb2f939d..b6c316f47cb5 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-core.c
@@ -95,6 +95,7 @@ struct dcmipp_pipeline_config {
 	size_t num_ents;
 	const struct dcmipp_ent_link *links;
 	size_t num_links;
+	u32 hw_revision;
 };
 
 /* --------------------------------------------------------------------------
@@ -130,11 +131,13 @@ static const struct dcmipp_ent_link stm32mp13_ent_links[] = {
 			MEDIA_LNK_FL_ENABLED | MEDIA_LNK_FL_IMMUTABLE),
 };
 
+#define DCMIPP_STM32MP13_VERR	0x10
 static const struct dcmipp_pipeline_config stm32mp13_pipe_cfg = {
 	.ents		= stm32mp13_ent_config,
 	.num_ents	= ARRAY_SIZE(stm32mp13_ent_config),
 	.links		= stm32mp13_ent_links,
-	.num_links	= ARRAY_SIZE(stm32mp13_ent_links)
+	.num_links	= ARRAY_SIZE(stm32mp13_ent_links),
+	.hw_revision	= DCMIPP_STM32MP13_VERR
 };
 
 #define	ID_MAIN_ISP 3
@@ -218,11 +221,13 @@ static const struct dcmipp_ent_link stm32mp25_ent_links[] = {
 			MEDIA_LNK_FL_ENABLED | MEDIA_LNK_FL_IMMUTABLE),
 };
 
+#define DCMIPP_STM32MP25_VERR	0x30
 static const struct dcmipp_pipeline_config stm32mp25_pipe_cfg = {
 	.ents		= stm32mp25_ent_config,
 	.num_ents	= ARRAY_SIZE(stm32mp25_ent_config),
 	.links		= stm32mp25_ent_links,
-	.num_links	= ARRAY_SIZE(stm32mp25_ent_links)
+	.num_links	= ARRAY_SIZE(stm32mp25_ent_links),
+	.hw_revision	= DCMIPP_STM32MP25_VERR
 };
 
 /* -------------------------------------------------------------------------- */
@@ -575,6 +580,7 @@ static int dcmipp_probe(struct platform_device *pdev)
 		sizeof(dcmipp->mdev.model));
 	snprintf(dcmipp->mdev.bus_info, sizeof(dcmipp->mdev.bus_info),
 		 "platform:%s", DCMIPP_PDEV_NAME);
+	dcmipp->mdev.hw_revision = pipe_cfg->hw_revision;
 	dcmipp->mdev.dev = &pdev->dev;
 	dcmipp->mdev.ops = &dcmipp_media_ops;
 	media_device_init(&dcmipp->mdev);
-- 
2.39.2

